src/html-imports.js

```diff
@@ -156,60 +156,52 @@
-        if (!url) {
-          reject({
-            resource: 'error: href must be specified'
-          });
-        } else if (url.match(/^data:/)) {
-          // Handle Data URI Scheme
-          const pieces = url.split(',');
-          const header = pieces[0];
-          let resource = pieces[1];
-          if (header.indexOf(';base64') > -1) {
-            resource = atob(resource);
-          } else {
-            resource = decodeURIComponent(resource);
-          }
-          resolve({
-            resource: resource
-          });
+    load(url, success, fail) {
+      if (!url) {
+        fail('error: href must be specified');
+      } else if (url.match(/^data:/)) {
+        // Handle Data URI Scheme
+        const pieces = url.split(',');
+        const header = pieces[0];
+        let resource = pieces[1];
+        if (header.indexOf(';base64') > -1) {
+          resource = atob(resource);
         } else {
-          const request = new XMLHttpRequest();
-          request.open('GET', url, Xhr.async);
-          request.addEventListener('readystatechange', () => {
-            if (request.readyState === 4) {
-              // Servers redirecting an import can add a Location header to help us
-              // polyfill correctly.
-              let redirectedUrl = undefined;
-              try {
-                const locationHeader = request.getResponseHeader('Location');
-                if (locationHeader) {
-                  // Relative or full path.
-                  redirectedUrl = (locationHeader.substr(0, 1) === '/') ?
-                    location.origin + locationHeader : locationHeader;
-                }
-              } catch (e) {
-                console.error(e.message);
-              }
-              const resp = {
-                resource: (request.response || request.responseText),
-                redirectedUrl: redirectedUrl
-              };
-              if (request.status === 304 || request.status === 0 ||
-                request.status >= 200 && request.status < 300) {
-                resolve(resp);
-              } else {
-                reject(resp);
+          resource = decodeURIComponent(resource);
+        }
+        success(resource);
+      } else {
+        const request = new XMLHttpRequest();
+        request.open('GET', url, Xhr.async);
+        request.addEventListener('readystatechange', () => {
+          if (request.readyState === 4) {
+            // Servers redirecting an import can add a Location header to help us
+            // polyfill correctly.
+            let redirectedUrl = undefined;
+            try {
+              const locationHeader = request.getResponseHeader('Location');
+              if (locationHeader) {
+                // Relative or full path.
+                redirectedUrl = (locationHeader.substr(0, 1) === '/') ?
+                  location.origin + locationHeader : locationHeader;
               }
+            } catch (e) {
```

Comment:
I originally found it there from V0, not sure either which part was failing, I assume `location.origin` was having some problems in IE https://connect.microsoft.com/IE/feedback/details/1763802/location-origin-is-undefined-in-ie-11-on-windows-10-but-works-on-windows-7
I've refactored it so that we don't need the try-catch